### PR TITLE
Fetch Discourse usernames from smart contract (#85)

### LINKED ISSUE #85: Fetch Discourse usernames from smart contract

### PR DESCRIPTION
**Pull request by aahna-ashina**
## Related GitHub Issue

closes #85

## How Has This Been Tested?

- [x] localhost: `npm run generate-csv`
- [x] Status checks pass
- [ ] Works on Goerli
- [x] Works on Mainnet

### CODE REVIEW

File: data-sources/discourse/src/generate-csv.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  const writer = csvWriter.createObjectCsvWriter({
+    path: outputFilePath,
+    header: [
+      { id: 'passport_id', title: 'passport_id' },
+      { id: 'owner_address', title: 'owner_address' },
+      { id: 'discourse_username', title: 'discourse_username' }
+    ]
+  })
+  const csvRows: any[] = []
```
Comment (johnmark13) [inline data-sources/discourse/src/generate-csv.ts:34]: At some point we should create a type for the Row, we have one anyway!

**Review by johnmark13 — APPROVED** [review]
Are we going this route, there was a discussion somewhere about *not* doing this and using another mechanism for storing the data *and* I think someone created a single contract to hold all username data?

File: data-sources/discourse/src/generate-csv.ts (hunk 1 above)
Comment (aahna-ashina) [reply data-sources/discourse/src/generate-csv.ts:34]: @johnmark13 Yes, created an issue for it here:  https://github.com/nation3/nationcred-datasets/issues/90

**Comment by aahna-ashina** [comment]
> Are we going this route, there was a discussion somewhere about _not_ doing this and using another mechanism for storing the data _and_ I think someone created a single contract to hold all username data?

@johnmark13 If someone wants to refactor the current version of the software, feel free to open a [pull request](https://github.com/nation3/nationcred-datasets/pulls) for it 😄

**Pull request merged**